_content/tour/ukr/pointers.article

```diff
@@ -183,114 +175,110 @@ No allocation.
+у стек, проте деякі все-таки можуть бути скопійовані, залежно від доцільності використання регістрів.
+Проте, це не міняє жодну з семантик описаних у цій главі.
 
-** Notes
+** Памʼять виділяється в кучі
 
-- When a value could be referenced after the function that constructs the value returns.
-- When the compiler determines a value is too large to fit on the stack.
-- When the compiler doesn’t know the size of a value at compile time.
-- When a value is decoupled through the use of function or interface values.
+- Коли на значення можуть посилатися після завершення виконання функції, в якій було створено це значення.
+- Коли компілятор розуміє що значення занадто велике, щоб вміститися в стеку.
+- Коли компілятор не знає розмір значення під час компіляції.
+- Коли значення відокремлено через використання типів інтерфейсів або функцій.
 
-** Garbage Collection Semantics
+** Семантика збірника сміття (Garbage Collection)
 
-Once a value is constructed on the heap, the Garbage Collector (GC) has to get
-involved. The most important part of the GC is the pacing algorithm. It determines
-the frequency/pace that the GC has to run in order to maintain the smallest heap
-possible in conjunction with the best application throughput.
+Коли значення створюється в кучі, збірник сміття вимушений долучатися.
+Найголовніша частина збірника сміття це алгоритм темпу. Він визначає частоту/темп
+з яким збірник сміття виконується, щоб підтримувати найменшу кількість значень у кучі, разом 
+з тим зберігати найкращу пропускну здатність програми.
 
 - [[https://www.ardanlabs.com/blog/2018/12/garbage-collection-in-go-part1-semantics.html][Garbage Collection Semantics Part I]] - William Kennedy
 
-** Stack vs Heap
+** Стек або куча
 
-"The stack is for data that needs to persist only for the lifetime of the function
-that constructs it, and is reclaimed without any cost when the function exits. The
-heap is for data that needs to persist after the function that constructs it exits,
-and is reclaimed by a sometimes costly garbage collection." - Ayan George
+"Стек потрібен для даних, яким потрібно зберігатися тільки протягом життєвого циклу функції,
+яка їх створює, і очищатися без жодних витрат, коли функція завершує виконання. Куча потрібна
+для даних, яким потрібно зберігатися навіть після завершення виконання функції, в якій вони створилися,
+і очищатися дорогим механізмом збірки сміття." - Ayan George
 
-** Stack Growth
+** Збільшення стеку
 
-The size of each frame for every function is calculated at compile time. This means,
-if the compiler doesn’t know the size of a value at compile time, the value must be
-constructed on the heap. An example of this is using the built-in function make to
-construct a slice whose size is based on a variable.
+Розмір кожного фрейму для кожної функції вираховується під час компіляції. Це значить,
+що якщо компілятор не знає розмір значення під час компіляції, то значення зобовʼязано
```

Comment:
https://slovnyk.ua/index.php?swrd=%D0%B7%D0%BE%D0%B1%D0%BE%D0%B2%27%D1%8F%D0%B7%D0%B0%D0%BD%D0%BE

```suggestion
що якщо компілятор не знає розмір значення під час компіляції, то значення зобовʼязане
```